synapse/handlers/sync.py

```diff
@@ -1401,9 +1404,12 @@ async def generate_sync_result(
                     mutable_joined_room_ids,
                 )
             )
-            for room_id in un_partial_stated_rooms:
-                if not await self.store.is_partial_state_room(room_id):
-                    forced_newly_joined_room_ids.add(room_id)
+            results = await self.store.is_partial_state_rooms(un_partial_stated_rooms)
+            forced_newly_joined_room_ids.update(
+                room_id
+                for room_id, is_partial_state in results.items()
+                if is_partial_state
```

Comment:
Should this be the other way around? The original code has a `not` in it.
```suggestion
                if not is_partial_state
```